### ISSUE TITLE
Performance using the Node API

### ISSUE THREAD

**Issue by oscarotero** [CloudCannon/pagefind#461]
Hi. I noticed a huge performance difference between CLI and NodeJS APIs.

Lume uses the JS API to index the pages before saving them into the disk, [as you can see here](https://github.com/lumeland/lume/blob/cf1e917a74210888b2e54b955be6ee91051276e6/plugins/pagefind.ts#L137-L149).

A site with less than 600 HTML files takes about **4 minutes** in my computer to be indexed. I noticed the first pages are indexed quickly but every new page added takes a bit longer than previous.

Using the CLI API, the site is indexed in just **2-3 seconds.**

Is there any way to improve this? Perhaps a function to send all pages to be indexed at the same time? For example:

```js
await index.addHTMLFiles([
  {
    url: "/page1/",
    content: "content1",
  },
  {
    url: "/page2/",
    content: "content2",
  },
  // etc
]);
```

**Comment by bglw** [comment]
Will look at this week. ([Some discussion in Lume discord](https://discord.com/channels/794537085641818124/1158469690499276850))

- [ ] Add plural files method(s) that can index in parallel, matching CLI parallelization
- [ ] Look into transferring large content though tmp files instead of stdio for performance 
- [ ] Investigate why indexing subsequent pages slows down as this feel unexpected based on the work being done

**Comment by bglw** [comment]
Found the first bug causing the biggest issue. If you update to `v1.0.4-rc0` performance should be way <sup>way <sup>way</sup></sup> better. Will look at the other tasks later, but they should be (less) critical now.

**Comment by bglw** [comment]
Released as stable in [v1.0.4](https://github.com/CloudCannon/pagefind/releases/tag/v1.0.4) 🎉
